app/src/main/java/com/example/palletify/ui/generator/GenerateWithImageScreen.kt

```diff
@@ -63,92 +130,119 @@ fun GenerateWithImageScreen(
+                                } else {
+                                    BorderStroke(0.dp, Color.White)
+                                }
+                            )
+                        ,
+                    )
                 }
-                Button(
-                    modifier = Modifier
-                        .padding(top = 8.dp)
-                        .heightIn(min = 55.dp),
-                    colors = ButtonDefaults.buttonColors(containerColor = Color(0xFF775798)),
-                    onClick = { /* No action is triggered */ }
-                ) {
-                    Text("")
+                if (colors.size <= 5) {
+                    Button(
+                        modifier = Modifier
+                            .size(60.dp)
+                            .heightIn(min = 55.dp),
+
+                        onClick = {
+                            if (moreColors.isNotEmpty()) {
+                                colors.add(moreColors[0])
+                                moreColors.removeAt(0)
+                            }
+                        }
+                    ) {
+                        Text(
+                            text = "+",
+                            style = TextStyle(
+                                fontSize = 32.sp, fontWeight = FontWeight.Medium
+                            ),
+                            textAlign = TextAlign.Center
+                        )
+                    }
                 }
+
             }
-            Button(modifier = Modifier
-                .padding(12.dp)
-                .heightIn(min = 10.dp)
-                .fillMaxSize(),
-                onClick = {
-                    val currentPalette = generatorViewModel.uiState.value.currentPalette
-                    val numberOfColors = generatorViewModel.uiState.value.numberOfColours
-                    val mode = generatorViewModel.uiState.value.mode
-                    val colorsList = mutableListOf<String>()
-                    for (i in 0 until numberOfColors) {
-                        val color = currentPalette[i].hex.value
-                        colorsList.add(color)
+            if (selectedOption == ""){
+                Button(modifier = Modifier
+                    .padding(12.dp)
+                    .heightIn(min = 55.dp)
+                    .fillMaxWidth(),
+                    onClick = {
+                        val currentPalette = generatorViewModel.uiState.value.currentPalette
+                        val numberOfColors = generatorViewModel.uiState.value.numberOfColours
+                        val mode = generatorViewModel.uiState.value.mode
+                        val colorsList = mutableListOf<String>()
+                        for (i in 0 until numberOfColors) {
+                            val color = currentPalette[i].hex.value
+                            colorsList.add(color)
+                        }
```

Comment:
i thought we needed the colours list to be a Color object which includes hex, rgb, and name to work with the main colour generator and preview data

Reply:
oops, i'll look into that

Reply:
oh i think its like this because Room can't store custom data types well (it can but u need to use type converters but for Color object I think its kind of complex), so I ended up just saving a list of each colours hex value 

ill look into if its possible to store Color objects in a list, but I think a workaround is we store just the hex, rgb and name separately and then recreate the Color object when you get it from the db? idk let me know what yall think

Reply:
sounds good ill check with them!